helper/daemon/daemon.go

```diff
@@ -0,0 +1,170 @@
+package daemon
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"time"
+)
+
+const EnvName = "XW_DAEMON_IDX"
+
+// The number of times background was called at runtime
+var runIdx int = 0
+
+// Daemon
+type Daemon struct {
+	// records the standard output and error output of daemons and child processes
+	LogFile string
+	// The maximum number of cycle restarts. If it is 0, it will restart indefinitely
+	MaxCount int
+	// The maximum number of consecutive startup failures or abnormal exits.
+	// Exceed this number, the daemon exits and the child process will not be restarted
+	MaxError int
+	// The minimum time (in seconds) for a child process to exit normally
+	// Less than this time is considered as abnormal exit
+	MinExitTime int64
+	// validator key for daemon
+	ValidatorKey string
+}
+
+// Turn the the main program into background running (start a child process, and then exit)
+// logFile - If it is not empty, the standard output and error output of the child process will be recorded in this file
+// isExit - Whether to exit the main program directly after the child process started.
+//          If false, the main program returns * OS Process, the child process returns nil.
+//          It needs to be handled by the caller.
+func Background(logFile string, priKey string, isExit bool) (*exec.Cmd, error) {
+	// Check whether the child process or the parent process
+	runIdx++
+
+	envIdx, err := strconv.Atoi(os.Getenv(EnvName))
```

Comment:
`len()` condition would be better understood.

Reply:
envIdx has to assgn value, no need use more variable.